src/Workspaces/Remote/ServiceHub/Services/UnusedReferences/ProjectAssets/RemoteProjectAssetsReaderService.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.IO;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.UnusedReferences;
+using Microsoft.CodeAnalysis.UnusedReferences.ProjectAssets;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Remote.UnusedReferences.ProjectAssets
+{
+    [ExportWorkspaceService(typeof(IProjectAssetsReaderService)), Shared]
+    internal sealed class RemoteProjectAssetsReaderService : IProjectAssetsReaderService
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public RemoteProjectAssetsReaderService()
+        {
+        }
+
+        public ImmutableArray<ReferenceInfo> ReadReferences(
+            ImmutableArray<ReferenceInfo> projectReferences,
+            string projectAssetsFilePath)
+        {
+            if (!File.Exists(projectAssetsFilePath))
+            {
+                return ImmutableArray<ReferenceInfo>.Empty;
+            }
+
+            var projectAssetsFileContents = File.ReadAllText(projectAssetsFilePath);
+            ProjectAssetsFile projectAssets;
+
+            try
+            {
+                projectAssets = JsonConvert.DeserializeObject<ProjectAssetsFile>(projectAssetsFileContents);
+            }
+            catch
+            {
+                return ImmutableArray<ReferenceInfo>.Empty;
+            }
+
+            return ProjectAssetsReader.EnhanceReferences(projectReferences, projectAssets);
```

Comment:
this code seems identical.  move to new common base class?

Reply:
The issue is finding a common project which a Newtonsoft.Json reference. I extracted all the common processing code into the ProjectAssetsReader.

Reply:
I've made this a static class that is linked to the two projects that need it.